### PR TITLE
Windows build & Azure CI

### PR DESCRIPTION
**Pull request by ojura**
Depends on googlecartographer/cartographer#1463

`cartographer_ros` should build out-of-box with [ROS for Windows](http://aka.ms/ros), 
following the more-or-less usual build instructions (need admin for `choco upgrade` and for `rosdep install`). Be sure to also have `call`-ed the appropriate Visual Studio environment `VsDevCmd.bat` or `vcvars64.bat` script. If you have git bash installed with Cygwin executables in PATH, be sure to `call cartographer\scripts\remove_mingw_cygwin_from_path.bat` before building.

    choco upgrade ros-melodic-desktop -y
    choco upgrade ros-melodic-perception -y
    mkdir C:\catkin_ws\src & cd C:\catkin_ws\src
    git clone https://github.com/googlecartographer/cartographer
    git clone https://github.com/googlecartographer/cartographer_ros
    cd C:\catkin_ws & call C:\opt\ros\melodic\x64\setup.bat
    rosdep install --from-paths src --ignore-src -r -y
    catkin_make_isolated --use-ninja --install

Tested with (picture below)

    call install_isolated\setup.bat
    roslaunch cartographer_ros offline_backpack_2d.launch \
              bag_filenames:=%HOMEPATH%/Downloads/cartographer_paper_deutsches_museum.bag

### CODE REVIEW

**Comment by ojura** [comment]
image

File: cartographer_rviz/CMakeLists.txt [hunk 1]
```diff
@@ -52,11 +56,9 @@ if(rviz_QT_VERSION VERSION_LESS "5")
 else()
   message(STATUS "Using Qt5 based on the rviz_QT_VERSION: ${rviz_QT_VERSION}")
   find_package(Qt5 ${rviz_QT_VERSION} EXACT REQUIRED Core Widgets)
-  set(QT_LIBRARIES Qt5::Widgets)
+  set(QT_LIBRARIES Qt5::Core Qt5::Widgets)
   include_directories(${Qt5Widgets_INCLUDE_DIRS})
 endif()
-add_definitions(-DQT_NO_KEYWORDS)
-add_library(${PROJECT_NAME} ${ALL_SRCS})
```
Comment (ojura) [inline cartographer_rviz/CMakeLists.txt]: Do we really need QT_NO_KEYWORDS? If so, consider making it a private definition?

Comment (gaschler) [reply cartographer_rviz/CMakeLists.txt]: Feel free to question and simplify the CMake build, but please in a separate change. This one is already complicated enough.

**Comment by ojura** [comment]
Okay, I merged #1126 and cleaned up the diff due to minor differences how this PR and #1126 fixed linking with Qt. Should be ready again.

**Comment by ojura** [comment]
Noticed one test failing:
https://dev.azure.com/larics/cartographer/_build/results?buildId=189&view=ms.vss-test-web.test-result-details

**Pull request merged**